crates/sui-config/src/node.rs

```diff
@@ -78,7 +78,7 @@ pub struct NodeConfig {
     pub grpc_concurrency_limit: Option<usize>,
 
     #[serde(default)]
-    pub p2p_config: P2pConfig,
+    pub p2p_config: Option<P2pConfig>,
```

Comment:
I don't think this is the correct way to turn off state-sync, since there may be other services running on this p2p network (eg discovery) that we may want to have running.